source/azure_iot_json_reader.c

```diff
@@ -0,0 +1,237 @@
+    {
+        AZLogError( ( "Could not skip children in JSON: core error=0x%08x", xCoreResult ) );
+        xResult = eAzureIoTErrorFailed;
+    }
+    else
+    {
+        xResult = eAzureIoTSuccess;
+    }
+
+    return xResult;
+}
+
+
+AzureIoTResult_t AzureIoTJSONReader_GetTokenBool( AzureIoTJSONReader_t * pxReader,
+                                                  bool * pxValue )
+{
+    AzureIoTResult_t xResult;
+    az_result xCoreResult;
+
+    if( ( pxReader == NULL ) || ( pxValue == NULL ) )
+    {
+        AZLogError( ( "AzureIoTJSONReader_GetTokenBool failed: invalid argument" ) );
+        return eAzureIoTErrorInvalidArgument;
+    }
+
+    if( az_result_failed( xCoreResult = az_json_token_get_boolean( &pxReader->_internal.xCoreReader.token, pxValue ) ) )
+    {
+        AZLogError( ( "Could not get boolean in JSON: core error=0x%08x", xCoreResult ) );
+        xResult = eAzureIoTErrorFailed;
+    }
+    else
+    {
+        xResult = eAzureIoTSuccess;
+    }
+
+    return xResult;
+}
+
+
+AzureIoTResult_t AzureIoTJSONReader_GetTokenInt32( AzureIoTJSONReader_t * pxReader,
+                                                   int32_t * pilValue )
+{
+    AzureIoTResult_t xResult;
+    az_result xCoreResult;
+
+    if( ( pxReader == NULL ) || ( pilValue == NULL ) )
+    {
+        AZLogError( ( "AzureIoTJSONReader_GetTokenInt32 failed: invalid argument" ) );
+        return eAzureIoTErrorInvalidArgument;
+    }
+
+    if( az_result_failed( xCoreResult = az_json_token_get_int32( &pxReader->_internal.xCoreReader.token, pilValue ) ) )
+    {
+        AZLogError( ( "Could not get int32_t in JSON: core error=0x%08x", xCoreResult ) );
+        xResult = eAzureIoTErrorFailed;
+    }
+    else
+    {
+        xResult = eAzureIoTSuccess;
+    }
+
+    return xResult;
+}
+
+
+AzureIoTResult_t AzureIoTJSONReader_GetTokenDouble( AzureIoTJSONReader_t * pxReader,
+                                                    double * pxValue )
+{
+    AzureIoTResult_t xResult;
+    az_result xCoreResult;
+
+    if( ( pxReader == NULL ) || ( pxValue == NULL ) )
+    {
+        AZLogError( ( "AzureIoTJSONReader_GetTokenDouble failed: invalid argument" ) );
+        return eAzureIoTErrorInvalidArgument;
+    }
+
+    if( az_result_failed( xCoreResult = az_json_token_get_double( &pxReader->_internal.xCoreReader.token, pxValue ) ) )
+    {
+        AZLogError( ( "Could not get double in JSON: core error=0x%08x", xCoreResult ) );
+        xResult = eAzureIoTErrorFailed;
+    }
+    else
+    {
+        xResult = eAzureIoTSuccess;
+    }
+
+    return xResult;
+}
+
+
+AzureIoTResult_t AzureIoTJSONReader_GetTokenString( AzureIoTJSONReader_t * pxReader,
+                                                    uint8_t * pucBuffer,
+                                                    uint32_t usBufferSize,
+                                                    uint32_t * pusBytesCopied )
```

Comment:
>pusBytesCopied [](http://example.com/codeflow?start=63&length=14)

ul?

Reply:
did a find all and replace everywhere for them